### PR TITLE
ncc unistore to fix install warning

### PR DESCRIPTION
**Pull request by Timer**
This pull request adds a new `precompile` step for `ncc`ing direct dependencies.

1. Added new behavior to `ncc()` if `packageName` option is specified
1. Moved target dependency to `devDependencies`
1. Changed usages to be `next/dist/compiled/<package>`

The new `ncc()` behavior emulates a nested package (defines a `package.json`).

--- 

This does **not** overwrite the imports (i.e. `module-alias`) because that normally breaks at scale.

### CODE REVIEW

File: packages/next/taskfile.js [hunk 1]
```diff
@@ -18,6 +21,18 @@ const babelOpts = {
   ]
 }
 
+export async function nccunistore (task, opts) {
+  await task
+    .source(opts.src || relative(__dirname, require.resolve('unistore')))
+    .ncc()
+    .target('dist/compiled/unistore')
+  writePackageManifest('unistore')
```
Comment (dav-is) [inline packages/next/taskfile.js]: How come you don't just do this by pushing a file in the plugin?
```
this._.files.push({ dir, base, data })
```

Comment (Timer) [reply packages/next/taskfile.js]: I went ahead and changed it.

Comment (Timer) [reply packages/next/taskfile.js]: I went ahead and changed it.

**Review by timneutkens — APPROVED** [review]

**Review by dav-is — APPROVED** [review]
Looks great besides `nccunistore` could use some camel case or a different name

**Pull request merged**